sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -115,6 +116,8 @@ public Mono<HttpResponse> send(HttpRequest request, Context context) {
             .filter(timeoutDuration -> timeoutDuration instanceof Duration)
             .map(timeoutDuration -> ((Duration) timeoutDuration).toMillis())
             .orElse(this.responseTimeout);
+        boolean effectiveHeadersEagerlyConverted = (boolean) context.getData(AZURE_EAGERLY_CONVERT_HEADERS)
```

Comment:
I wonder if per-request granularity is needed? Otherwise, it would be more efficient to have a property on the HttpClientOptions.